packages/tools-node/src/package.ts

```diff
@@ -0,0 +1,212 @@
+import { pickValues } from "@rnx-kit/tools-language/props";
+import findUp from "find-up";
+import fs from "fs";
+import path from "path";
+import pkgDir from "pkg-dir";
+import pkgUp from "pkg-up";
+
+/**
+ * Components of a package reference.
+ */
+export type PackageRef = {
+  scope?: string;
+  name: string;
+};
+
+/**
+ * Parse a package reference string. One exaple is `name` property found in `package.json`.
+ *
+ * @param r Package reference string
+ * @returns Parsed package reference object
+ */
+export function parsePackageRef(r: string): PackageRef {
+  if (r.startsWith("@")) {
+    const indexSeparator = r.indexOf("/");
+    // The separator must start from position >= 2 to ensure that it is '@'
+    // and at least one other character. Further, the separator must have
+    // at least 1 character following it, before the end of the string.
+    if (indexSeparator < 2 || indexSeparator + 2 >= r.length) {
+      throw new Error(`Invalid package reference: "${r}"`);
+    }
+    return {
+      scope: r.substring(0, indexSeparator),
+      name: r.substring(indexSeparator + 1),
+    };
+  }
+
+  if (!r) {
+    throw new Error(`Invalid package reference: "${r}"`);
+  }
+  return { name: r };
+}
+
+/**
+ * Get the mangled name for a package reference.
+ *
+ * @param ref Package reference
+ * @returns Mangled name
+ */
+export function getMangledPackageName(ref: PackageRef): string {
+  if (ref.scope) {
+    return ref.scope.slice(1) + "__" + ref.name;
+  }
+  return ref.name;
+}
+
+/**
+ * Schema for a reference to a person in `package.json`.
+ */
+export type PackagePerson =
```

Comment:
Shouldn't this be author?
```suggestion
export type PackageAuthor =
```

Reply:
PackagePerson will also apply to contributors as PackagePerson[]. I didn't add that field, though, because no one was using it.